src/views/AddItem.jsx

```diff
@@ -1,3 +1,83 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { React, useState } from 'react';
+
+import { addItem } from '../api/firebase';
+
+//we need listToken (listID) passing from App.js
+
+export function AddItem({ listToken }) {
+	const [itemName, setItemName] = useState('');
+	const [daysUntilNextPurchase, setDaysUntilNextPurchase] = useState(7);
+
+	const submitForm = (e) => {
+		e.preventDefault();
+		try {
+			addItem(listToken, { itemName, daysUntilNextPurchase });
+			alert('Item is saved in the database.');
+		} catch (error) {
+			alert('Item is not saved in the database.');
+		}
+		setItemName('');
+		setDaysUntilNextPurchase(7);
+	};
+
+	const handleNameInput = (e) => setItemName(e.target.value);
+
+	const handleFrequencyInput = (e) => setDaysUntilNextPurchase(e.target.value);
+
+	return (
+		<form onSubmit={(e) => submitForm(e)}>
+			<fieldset>
+				<label htmlFor="itemName">Item name:</label>
+				<input
+					type="text"
+					id="itemName"
+					name="itemName"
+					onChange={handleNameInput}
+					required
+				/>
+
+				<fieldset>
+					<label htmlFor="itemFrequencyInDays">
+						How soon will you buy this again?
+					</label>
+					<input
+						type="radio"
+						id="7"
+						name="itemFrequencyInDays"
+						value="7"
+						defaultChecked
```

Comment:
If we change `defaultChecked required` to  `checked={daysUntilNextPurchase === 7}`
then when the user adds a daysUntilNextPurchase value of 14 or 21 days, the form will reset back to 7 days.